### PR TITLE
Take TX out of RX mode when doing config commit

### PR DESCRIPTION
**Pull request by CapnBry**
Before doing a `config.commit()` on the TX, take the RF out of receive mode. If a packet comes in while we're spinning on the nvs commit, it will cause a crash as we try to process the ISR.

Ref #1394 

I believe the order needs to be
* switch callbackTock
* disable RX

This is because we could start disabling the RX, then instantly get the tock ISR before the handler changes, which would put the RF back into RX mode if it was supposed to happen next and then the problem would still occur.

I also had to go back to tracking all three phases of the telem receiver: 
1. TX mode
2. After TX done, before Tock (switched to receive mode but still before the actual expected receive slot)
3. During the RX slot. Tock has fired and we're in RX mode, expecting that the RF is actually doing a receive or has recently finished it.

In the first version of this using the existing code where it only tracked Phase 1 and Phase 2, the module is in RX mode which needs to be disabled (the reason for this PR) but it only can tell to make the switch if it is in Phase 2, which only lasts ~400us of the 2400us period. If we don't track Phase 3 as well, the alternative is to always call `Radio.SetTxIdleMode()` to force it out. I feel like tracking all 3 phases is better long term so we can actually tell which we're in. 

### Testing
For anyone who wants to confirm this bug on ESP32 TX, connect with telemetry running at 1:2 500Hz to get the most data coming back to the TX. For me it takes 3-12 tries on master before the module reboots itself, it doesn't matter if it is a change from the OLED+Joystick or the Lua, any config change will do it.

Tested on BetaFPV 1W TX for all development, then tested on FM30 for compatibility.

### CODE REVIEW

File: src/src/tx_main.cpp [hunk 1]
```diff
@@ -603,13 +602,15 @@ static void CheckConfigChangePending()
...
-    // If telemetry expected in the next interval, the radio is in RX mode
-    // and will skip sending the next packet when the tiemr resumes.
+    // If telemetry expected in the next interval, the radio was in RX mode
+    // and will skip sending the next packet when the timer resumes.
     // Return to normal send mode because if the skipped packet happened
     // to be on the last slot of the FHSS the skip will prevent FHSS
     if (TelemetryRcvPhase == ttrpInReceiveMode)
     {
+      Radio.SetTxIdleMode();
       TelemetryRcvPhase = ttrpTransmitting;
```
Comment (JyeSmith) [inline src/src/tx_main.cpp:614]: Any reason not to add `TelemetryRcvPhase == ttrpTransmitting` to the above `if` condition?  Then this `if (TelemetryRcvPhase == ttrpInReceiveMode)` section can be removed and we remove an additional spi call.

`if ((config.IsModified() || ModelUpdatePending) && TelemetryRcvPhase == ttrpTransmitting)`

Comment (CapnBry) [reply src/src/tx_main.cpp:615]: No we can't test it before we disconnect the Tock handler. The ISR can change `TelemetryRcvPhase` after it is tested and our code will start to run expecting that it is in TX mode when by the time we start making changes it is already in RX mode.

**Review by AlessandroAU — APPROVED** [review]

**Review by pkendall64 — APPROVED** [review]

**Comment by CapnBry** [comment]
Please hold on merging this until after the LBT #1243 since this will create a conflict I can fix up once it is merged.

**Comment by CapnBry** [comment]
Wow, that merged cleaner than I thought it would. I thought for sure it was going to choke on the ttrp enum name change. Thanks, @pkendall64 this is ready to merge I think.

**Pull request merged**